lib/core/src/Cardano/Wallet/Primitive/Types/Tx.hs

```diff
@@ -465,18 +465,19 @@ cardanoTxFromBytes maxEra bs = asum $ map snd $ filter (withinEra maxEra . fst)
         ([], []) -> error "impossible"
 
 withinEra :: AnyCardanoEra -> AnyCardanoEra -> Bool
-withinEra (AnyCardanoEra AlonzoEra) _ = True
-withinEra (AnyCardanoEra MaryEra) (AnyCardanoEra AlonzoEra) = False
-withinEra (AnyCardanoEra MaryEra) (AnyCardanoEra _) = True
-withinEra (AnyCardanoEra AllegraEra) (AnyCardanoEra AlonzoEra) = False
-withinEra (AnyCardanoEra AllegraEra) (AnyCardanoEra MaryEra) = False
-withinEra (AnyCardanoEra AllegraEra) (AnyCardanoEra _) = True
-withinEra (AnyCardanoEra ShelleyEra) (AnyCardanoEra AlonzoEra) = False
-withinEra (AnyCardanoEra ShelleyEra) (AnyCardanoEra MaryEra) = False
-withinEra (AnyCardanoEra ShelleyEra) (AnyCardanoEra AllegraEra) = False
-withinEra (AnyCardanoEra ShelleyEra) (AnyCardanoEra _) = True
-withinEra (AnyCardanoEra ByronEra) (AnyCardanoEra ByronEra) = True
-withinEra (AnyCardanoEra ByronEra) _ = False
+withinEra (AnyCardanoEra e1) (AnyCardanoEra e2) = case (e1, e2) of
+    (AlonzoEra , _         ) -> True
+    (MaryEra   , AlonzoEra ) -> False
+    (MaryEra   , _         ) -> True
+    (AllegraEra, AlonzoEra ) -> False
+    (AllegraEra, MaryEra   ) -> False
+    (AllegraEra, _         ) -> True
+    (ShelleyEra, AlonzoEra ) -> False
+    (ShelleyEra, MaryEra   ) -> False
+    (ShelleyEra, AllegraEra) -> False
+    (ShelleyEra, _         ) -> True
+    (ByronEra  , ByronEra  ) -> True
+    (ByronEra  , _         ) -> False
```

Comment:
Hey @paweljakubas. I wonder if this is possible to express with `<=`.

Is there an `Ord` instance for the type of value within `AnyCardanoEra`?

If so, it seems like the case expression could be replaced with something like `x >= y`.

But perhaps I've misunderstood something here. 😄